addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/spider/OpenApiSpiderFunctionality.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.openapi.spider;
+
+import java.util.Locale;
+import net.htmlparser.jericho.Source;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.openapi.converter.Converter;
+import org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter;
+import org.zaproxy.zap.extension.openapi.network.Requestor;
+import org.zaproxy.zap.model.ValueGenerator;
+
+public class OpenApiSpiderFunctionality {
+
+    private static final Logger LOGGER = LogManager.getLogger(OpenApiSpiderFunctionality.class);
+    private static Requestor requestor;
```

Comment:
This is no longer being initialised, worth adding a test to ensure it's working properly.

Reply:
I re-added the initialization. Once the current test challenges are resolved, I can look into a test for this.

I looked quickly at just testing the constructor but there doesn't seem to be a way to `verify` the assignments within the constructor.

Reply:
Right, you would have to use it (i.e. call `parseResource`, it will be like the other "import" tests so can be based on those).

Reply:
I guess I shouldn't have tried this (PR), I still can't sort this out.

Reply:
This should do it:
```java
package org.zaproxy.zap.extension.openapi.spider;

import static fi.iki.elonen.NanoHTTPD.newFixedLengthResponse;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.contains;
import static org.mockito.Mockito.mock;

import fi.iki.elonen.NanoHTTPD.IHTTPSession;
import fi.iki.elonen.NanoHTTPD.Response;
import java.util.ArrayList;
import java.util.List;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.parosproxy.paros.network.HttpMessage;
import org.zaproxy.zap.extension.openapi.AbstractServerTest;
import org.zaproxy.zap.model.ValueGenerator;
import org.zaproxy.zap.testutils.NanoServerHandler;

/** Unit test for {@link OpenApiSpiderFunctionality}. */
class OpenApiSpiderFunctionalityUnitTest extends AbstractServerTest {

    private ValueGenerator valueGenerator;
    private OpenApiSpiderFunctionality spider;

    @BeforeEach
    void setupSpider() {
        valueGenerator = mock(ValueGenerator.class);
        spider = new OpenApiSpiderFunctionality(valueGenerator);
    }

    @Test
    void shouldParseResource() throws Exception {
        // Given
        List<String> accessedUris = new ArrayList<>();
        this.nano.addHandler(
                new NanoServerHandler("") {
                    @Override
                    protected Response serve(IHTTPSession session) {
                        accessedUris.add(session.getUri());
                        return newFixedLengthResponse("");
                    }
                });
        HttpMessage message = getHttpMessage("");
        message.setResponseBody(
                "openapi: 3.0.0\n"
                        + "servers:\n"
                        + "  - url: http://localhost:"
                        + nano.getListeningPort()
                        + "\n"
                        + "paths:\n"
                        + "  /path:\n"
                        + "    get:\n"
                        + "      responses:\n"
                        + "        200:\n"
                        + "          content:\n"
                        + "            application/json: {}");
        // When
        spider.parseResource(message, null, 0);
        // Then
        assertThat(accessedUris, contains("/path"));
    }
}
```

Reply:
TYVM @thc202